nearby-app/README.md

```diff
@@ -100,22 +108,145 @@ if (degrees > -22.5  && degrees <= 22.5){
+
+Clicking on the map icon will display the nearby places in the map view.
+
+## Displaying Places in the Map
+### Deriving a Viewpoint
+When the user clicks on the map icon to view the map, the map viewpoint is derived based on the locations in the list view.  This is done by iterating over the found places, creating a Multipoint object, and then using the GeometryEngine's buffer method to generate a polygon.  From the polygon, a rectangular area with a spatial reference can be obtained and passed to the map view.
 
 ```java
+List<Point> points = new ArrayList<>();
 
-// Setting GeocodeParameters (see the section above about Geocoding With Categories)
+// Create an array of Point objects based on place locations
+for ( Place place : places){
+     points.add(place.getLocation());
+}
+// Create a Multipoint object and then buffer it to create a polygon
+Multipoint mp = new Multipoint(points);
+Polygon polygon = GeometryEngine.buffer(mp, 0.0007);
 
-// Setting the restriction attributes for walk times
-// when solving for a route.
+// Get the envelope from the polygon
+Envelope viewpoint = polygon.getExtent();
+```
+### Location Display
+Once we have a MapView, we can set the viewpoint and leverage the SDK's LocationDisplay.  In version 100.00 of the Android SDK there's a bug when setting the viewpoint of the map view and turning on location display.  For example, the following code will not zoom to the given viewpoint.  Instead, the map extent remains unchanged when displaying device location.
+
+```
+mMapView = (MapView) root.findViewById(R.id.map);
+final ArcGISMap map = new ArcGISMap(Basemap.createNavigationVector());
+mMapView.setMap(map);
+
+// Setting the viewpoint and then
+// calling getLocationDisplay will cause
+// the map view to ignore the given viewpoint.
+
+mMapView.setViewpoint(mViewpoint); // a non-null viewpoint
+mLocationDisplay = mMapView.getLocationDisplay();
+mLocationDisplay.startAsync();
+
+```
+The desired behavior is to have the map view change the visible area of the map view to the given viewpoint and display the device location.  This is accomplished by the following steps:
+
+  1.  Setting the viewpoint
+  2.  Waiting for the map's draw status to be complete
+  3.  Getting the location display and starting it asynchronously
+
+```
+mMapView = (MapView) root.findViewById(R.id.map);
+ArcGISMap map = new ArcGISMap(Basemap.createNavigationVector());
+mMapView.setMap(map);
+
+// Set view point first
+mMapView.setViewpoint(mViewpoint);
+
+// Wait for draw status to be complete before getting and
+// starting location display.
+mMapView.addDrawStatusChangedListener(new DrawStatusChangedListener() {
+   @Override public void drawStatusChanged(final DrawStatusChangedEvent drawStatusChangedEvent) {
+        if (drawStatusChangedEvent.getDrawStatus() == DrawStatus.COMPLETED){
+          mLocationDisplay = mMapView.getLocationDisplay();
+          mLocationDisplay.startAsync();
+        }
+    }
+});
+```
```

Comment:
@doneill One last look to make sure I've correctly implemented your change request?